fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/rules/LogicalToPhysicalScanRule.java

```diff
@@ -45,18 +45,21 @@ public LogicalToPhysicalScanRule() {
         super(root);
     }
 
-    @Nonnull
     @Override
-    public ChangesMade onMatch(@Nonnull PlannerRuleCall call) {
+    public void onMatch(@Nonnull PlannerRuleCall call) {
         final IndexEntrySourceScanExpression logical = call.get(root);
         final IndexEntrySource indexEntrySource = logical.getIndexEntrySource();
 
         if (indexEntrySource.isIndexScan()) {
-            call.yield(call.ref(new RecordQueryIndexPlan(indexEntrySource.getIndexName(), logical.getScanType(),
-                    logical.getComparisons().toScanComparisons(), logical.isReverse())));
+            // If fields after we stopped comparing create duplicates, they might be empty, so that a record
+            // that otherwise matches the comparisons would be absent from the index entirely.
+            // We only know that we are done matching predicates when we convert to a physical scan, so we check here.
```

Comment:
Is there a test for this case? It seems tricky enough that we'd want it exercised somewhere.

Reply:
Yes. I believe we have several tests that use this, but at the very least `FDBRepeatedFieldQueryTest.textPrefixRepeated()` tests for this (which is how I noticed this issue in the new planner).